src/libraries/System.Reflection.Metadata/src/System/Reflection/Metadata/TypeNameParserHelpers.cs

```diff
@@ -388,5 +394,18 @@ internal static void ThrowInvalidOperation_HasToBeArrayClass()
             throw new InvalidOperationException();
 #endif
         }
+
+        internal static bool IsMaxDepthExceeded(TypeNameParseOptions options, int depth)
```

Comment:
The `TypeNameParserHelpers.cs` file is included as a link in the test project, so we can have plenty of small unit tests for each of the helper methods: https://github.com/dotnet/runtime/blob/main/src/libraries/System.Reflection.Metadata/tests/Metadata/TypeNameParserHelpersTests.cs

Because of that, it can not use internal methods of `TypeNameParseOptions`. That is why I've moved `IsMaxDepthExceeded` from `TypeNameParseOptions` to here.